app/Models/Farmasi/ResepObat.php

```diff
@@ -113,23 +132,11 @@ public function scopeKunjunganResep(Builder $query, string $jenisResep = 'umum',
             ->join('penjab', 'reg_periksa.kd_pj', '=', 'penjab.kd_pj')
             ->join('dokter', 'resep_obat.kd_dokter', '=', 'dokter.kd_dokter')
             ->join('poliklinik', 'reg_periksa.kd_poli', '=', 'poliklinik.kd_poli')
-            ->whereBetween('resep_obat.tgl_perawatan', [$tglAwal, $tglAkhir])
-            ->where('resep_obat.tgl_perawatan', '>', '0000-00-00')
-            ->when($shift === 'Pagi', function ($q) {
-                $q->whereBetween('resep_obat.jam', ['07:00:01', '14:00:00']);
+            ->where(function ($q) use ($waktuAwal, $waktuAkhir, $waktuAwalAkhir, $waktuAkhirAkhir) {
+                $q->whereBetween(DB::raw("concat(resep_obat.tgl_perawatan, ' ', resep_obat.jam)"), [$waktuAwal, $waktuAkhir])
+                ->orWhereBetween(DB::raw("concat(resep_obat.tgl_perawatan, ' ', resep_obat.jam)"), [$waktuAwalAkhir, $waktuAkhirAkhir]);
```

Comment:
Kondisi disini cuma munculkan shift di tanggal awal dan tanggal akhir saja

Reply:
```suggestion
                $q->whereBetween(DB::raw("concat(resep_obat.tgl_perawatan, ' ', resep_obat.jam)"), [
                    $tglAwal->setTimeFromTimeString($waktuShift->jam_masuk),
                    $tglAwal->setTimeFromTimeString($waktuShift->jam_pulang),
                ]);
            
                while ($tglAwal->lessThan($tglAkhir)) {
                    $tglAwal = $tglAwal->addDay();
                
                    $jamMasuk = $tglAwal->setTimeFromTimeString($waktuShift->jam_masuk);
                    $jamPulang = $shift === 'Malam'
                        ? $tglAwal->addDay()->setTimeFromTimeString($waktuShift->jam_pulang)
                        : $tglAwal->setTimeFromTimeString($waktuShift->jam_pulang);
                
                    $q->orWhereBetween(DB::raw("concat(resep_obat.tgl_perawatan, ' ', resep_obat.jam)"), [$jamMasuk, $jamPulang]);
                }    
```

saya mencobanya seperti ini dan berhasil

Reply:
Terlalu ribet, dipisah aja kondisi tanggal dan jamnya.
Kalau shift malam baru `$tglAkhir->addDay()`